Nef_3/include/CGAL/Nef_3/SNC_point_locator.h

```diff
@@ -223,220 +230,203 @@ class SNC_point_locator_by_spatial_subdivision :
-                      continue;
-                  eor = q;
-                  result = make_object(e);
-                  hit = true;
-                  _CGAL_NEF_TRACEN("the edge becomes the new hit object");
-              }
+            if( SNC_intersection::does_intersect_internally( ray, Segment_3(e->source()->point(),
+                                                                            e->twin()->source()->point()), q)) {
+              _CGAL_NEF_TRACEN("ray intersects edge on "<<q);
+              _CGAL_NEF_TRACEN("prev. intersection? "<<hit);
+              CGAL_assertion_code
+                  (if( hit) _CGAL_NEF_TRACEN("prev. intersection on "<<eor));
+              if( hit && !has_smaller_distance_to_point( ray.source(), q, eor))
+                continue;
+              _CGAL_NEF_TRACEN("is the intersection point on the current cell? "<<
+                               candidate_provider->is_point_in_node( q, n));
+              if( !candidate_provider->is_point_in_node( q, n))
+                continue;
+              eor = q;
+              e_res = e;
+              solution = is_edge_;
+              hit = true;
+              _CGAL_NEF_TRACEN("the edge becomes the new hit object");
+            }
           }
         }
-        else if( CGAL::assign( f, *o) && ((mask&4) != 0)) {
+      }
+      if((mask&4)!=0) {
+        for(typename Halffacet_list::const_iterator fi=n->facets_begin(); fi!=n->facets_end(); ++fi) {
+          Halffacet_handle f(*fi);
           Point_3 q;
           _CGAL_NEF_TRACEN("trying facet with on plane "<<f->plane()<<
-                  " with point on "<<f->plane().point());
+                           " with point on "<<f->plane().point());
           if( SNC_intersection::does_intersect_internally( ray, f, q) ) {
             _CGAL_NEF_TRACEN("ray intersects facet on "<<q);
             _CGAL_NEF_TRACEN("prev. intersection? "<<hit);
             if( hit) { _CGAL_NEF_TRACEN("prev. intersection on "<<eor); }
             if( hit && !has_smaller_distance_to_point( ray.source(), q, eor))
               continue;
             _CGAL_NEF_TRACEN("is the intersection point on the current cell? "<<
-                    candidate_provider->is_point_on_cell( q, objects_iterator));
-            if( !candidate_provider->is_point_on_cell( q, objects_iterator))
+                             candidate_provider->is_point_in_node( q, n));
+            if( !candidate_provider->is_point_in_node( q, n))
               continue;
             eor = q;
-            result = make_object(f);
+            f_res = f;
+            solution = is_facet_;
             hit = true;
             _CGAL_NEF_TRACEN("the facet becomes the new hit object");
           }
         }
-        else if((mask&15) == 15)
-          CGAL_error_msg( "wrong handle");
       }
       if(!hit)
-        ++objects_iterator;
+        ++nodes_iterator;
     }
+
     CGAL_NEF_TIMER(rs_t.stop());
-    return result;
+    switch (solution) {
```

Comment:
Yesterday I added the fourth `case`   as we got a warning not to have tested all cases. That now leads to a warning that not all control paths return a value.  Shall I put a `break` in the last case and the `return` statement after the switch?   @lrineau @sloriot @mglisse

Reply:
The usual would be `default: CGAL_UNREACHABLE();`, or possibly another `CGAL_UNREACHABLE();` after `}`, depending on what the compiler is complaining about. But I am a bit surprised by the warning you mention about not all control paths returning a value.

Reply:
The enum is in line 233 above and the switch lists them all.  And several compileres warn about it.

Reply:
> The enum is in line 233 above and the switch lists them all.

In C++, an enum can take values other than just those that were listed in its declaration. On the other hand, 4 values means exactly all values with 2 bits, so that should be the whole range.

Reply:
`default: break;` could also work. It could also be a series of `if/else if` ....